src/sparse_blas/backends/mklcpu/mklcpu_operations.cpp

```diff
@@ -0,0 +1,209 @@
+/*******************************************************************************
+* Copyright 2023 Codeplay Software Ltd.
+*
+* Licensed under the Apache License, Version 2.0 (the "License");
+* you may not use this file except in compliance with the License.
+* You may obtain a copy of the License at
+*
+* http://www.apache.org/licenses/LICENSE-2.0
+*
+* Unless required by applicable law or agreed to in writing,
+* software distributed under the License is distributed on an "AS IS" BASIS,
+* WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+* See the License for the specific language governing permissions
+* and limitations under the License.
+*
+*
+* SPDX-License-Identifier: Apache-2.0
+*******************************************************************************/
+
+// MKLCPU backend includes
+// These includes define their own oneapi::mkl::sparse namespace with some of the types that are used here: matrix_handle_t, index_base.
+#include <oneapi/mkl/spblas.hpp>
+
+#include "oneapi/mkl/sparse_blas/detail/helper_types.hpp"
+#include "oneapi/mkl/sparse_blas/detail/mklcpu/onemkl_sparse_blas_mklcpu.hpp"
+
+// Include are set up so that oneapi::mkl::sparse::matrix_handle_t refers to the MKLCPU handle and
+// oneapi::mkl::sparse::detail::matrix_handle* refers to the oneMKL interface handle in this file.
+
+namespace oneapi::mkl::sparse::mklcpu {
+
+sycl::event optimize_gemv(sycl::queue& /*queue*/, transpose /*transpose_val*/,
+                          detail::matrix_handle* /*handle*/,
+                          const std::vector<sycl::event>& /*dependencies*/) {
+    throw unimplemented("SPARSE_BLAS", "optimize_gemv");
```

Comment:
I know throwing exceptions was going to be the first step, but in case of `mklcpu` and/or `mklgpu` backends at least, I suppose we'll eventually just directly call something like:
```cpp
    return optimize_gemv(queue, transpose_val, detail::get_handle(handle), dependencies);
```
Presumably this will look very different for other backends, maybe some having intermediate steps and implementations to get to the final backend library call?

Reply:
I wasn't suggesting anything here. Just asking a question if, when we stop throwing exceptions for backends, for oneMKL backend, this will be a one-liner change from `throw` to `optimize_gemv()` call. Likewise, I am asking if for other backends, the change will involve more steps than calling just optimize_gemv() at this location or at some other location in a deeper layer.

Reply:
Ok, as you said for oneMKL it will be a one-liner call to `optimize_gemv`. For other backends ideally it should just launch a `host_task` to call a similar function.